saml.go

```diff
@@ -0,0 +1,98 @@
+package tfe
+
+import (
+	"context"
+)
+
+// Compile-time proof of interface implementation.
+var _ Samls = (*saml)(nil)
+
+// Samls describes SAML settings
+// TFE API docs: https://www.terraform.io/docs/cloud/api/agents.html
+type Samls interface {
```

Comment:
I'm keeping with the convention of plurasizing the interface, but presumably this could be changed to something better 😅